pkg/controller/component/component_controller.go

```diff
@@ -0,0 +1,252 @@
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource Component
+	err = c.Watch(&source.Kind{Type: &componentsv1alpha1.Component{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to secondary resource Pods and requeue the owner Component
+	//err = c.Watch(&source.Kind{Type: &corev1.Pod{}}, &handler.EnqueueRequestForOwner{
+	//	IsController: true,
+	//	OwnerType:    &componentsv1alpha1.Component{},
+	//})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+var _ reconcile.Reconciler = &ReconcileComponent{}
+var buildTyppeImages = map[string]string{"nodejs":"nodeshift/centos7-s2i-nodejs:10.x"}
+
+// ReconcileComponent reconciles a Component object
+type ReconcileComponent struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile reads that state of the cluster for a Component object and makes changes based on the state read
+// and what is in the Component.Spec
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileComponent) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling Component")
+
+	// Fetch the Component instance
+	instance := &componentsv1alpha1.Component{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	log.Info("============================================================")
+	log.Info(fmt.Sprintf("***** Reconciling Component %s, namespace %s", request.Name, request.Namespace))
+	log.Info(fmt.Sprintf("** Creation time : %s", instance.ObjectMeta.CreationTimestamp))
+	log.Info(fmt.Sprintf("** Resource version : %s", instance.ObjectMeta.ResourceVersion))
+	log.Info(fmt.Sprintf("** Generation version : %s", strconv.FormatInt(instance.ObjectMeta.Generation, 10)))
+	log.Info(fmt.Sprintf("** Deletion time : %s", instance.ObjectMeta.DeletionTimestamp))
+	log.Info("============================================================")
+
+	// Assign the generated ResourceVersion to the resource status
+	if instance.Status.RevNumber == "" {
+		instance.Status.RevNumber = instance.ObjectMeta.ResourceVersion
+	}
+
+	if !instance.ObjectMeta.DeletionTimestamp.IsZero() {
```

Comment:
If the object is deleted, we should end the reconciliation.

Reply:
cb16036